File: DSAUni/Python Conversion/linkedlist.py


class Linknode:
    def __init__(self,data):
        self.data = data
        self.next = None

    def insert(self,data):
        x = Linknode(data)
        x.next = self.next
        self.next = x
    def search(self,val):
        prev = None
        curr = self
        while curr is not None:
            if curr.data == val:
                return True,prev,curr
            prev = curr
            curr = curr.next
        return False,prev,curr

    def deletebyVal(self,val):
        item = None
        a = self
        found,prev,curr = a.search(val)
        if found is not True:
            return item,self
        if prev is None:
            item = curr.data
            return item, self.next
        item = curr.data
        prev.next = curr.next
        

        return item, self


a = Linknode(5)
a.insert(7)
a.insert(3)
a.insert(9)
b = a
while b is not None:
    
    print(b.data)
    b = b.next
print("Searching value")
value = a.search(7)
print("hello")
if value[0] is True:
    print(value[1].data)

    print(value[2].data)

delete, a = a.deletebyVal(3)
delete, a = a.deletebyVal(5)
delete, a = a.deletebyVal(7)

print("deleting value the value: ", delete)
while a is not None:
    
    print(a.data)
    a = a.next

File: DSAUni/C++/LinkedList/linkedlist.cpp
#include<iostream>



using namespace std;

class Linknode{
public:
    int data;
    Linknode *next;
    
    Linknode(int val){
        data = val;
        next = nullptr;
    }

    void insert(int val){
        Linknode *x = new Linknode(val);
        x->next = this->next;
        this->next = x;
    }
    


};

int main(){

Linknode *x = new Linknode(5);
x->insert(8);
x->insert(22);
x->insert(43);
x->insert(99);
x->insert(44);
Linknode *y = x;
while (y->next != nullptr)
{
    cout << y->data <<" ";
    y = y->next; 
}




}